src/librustc_resolve/diagnostics.rs

```diff
@@ -860,6 +860,26 @@ match (A, B, C) {
 ```
 "##,
 
+E0422: r##"
+You are trying to use an identifier that is either undefined or not a struct.
```

Comment:
I mostly kept status quo regarding error codes, it may make sense to have a separate code for each `(path_source, has_resolution)` combination, 3 new error codes will need to be allocated.